File: dataset/human/python/file_2961.py
from collections import Counter
N = int(input())
 
def prime_factorization(n):
    table = []
    for x in range(2, int(n ** 0.5) + 1):
        while n % x == 0:
            table.append(x)
            n //= x
    if n > 1:
        table.append(n)
    return table
 
T = prime_factorization(N)
C = Counter(T)
 
cnt = 0
for v in C.values():
    next = 1
    while v >= next:
        v -= next
        cnt += 1
        next += 1
 
print(cnt)

File: dataset/human/c/file_2961.c
#include <stdio.h>

int main(void)
{
	int k,s;
	scanf("%d %d",&k,&s);
	int x,y,z,cnt=0;
	
	for(x=0;x<=k;x++){
		if(0<=s-x && s-x<=2*k){
			for(y=0;y<=k;y++){
				if(s-x-y<=k && 0<=s-x-y){
					cnt++;
				}
			}
		}
	}
	printf("%d\n",cnt);
	return 0;
}
